src/main/java/com/github/davidcarboni/restolino/json/Serialiser.java

```diff
@@ -119,10 +135,40 @@ public static void serialise(OutputStream output, Object responseMessage)
      * @throws IOException If an error occurs in writing the output.
      */
     public static void serialise(Path output, Object json) throws IOException {
+        Path temp = null;
+        try {
+            temp = Files.createTempFile(json.getClass().getSimpleName(), ".json");
+            writeJsonObjectToFile(temp, output, json);
+        } finally {
+            deleteTempFile(temp);
+        }
+    }
+
+    private static void deleteTempFile(Path temp) throws IOException {
+        if (temp == null) {
+            LOG.debug("temp file is null no clean up required");
+            return;
+        }
+
+        LOG.debug("attempting to delete temp file created by serialise method file:{}", temp.toString());
+        if (!Files.deleteIfExists(temp)) {
+            LOG.warn("attempt to delete temp file was unsuccessful, file has been added to deleteOnExit list " +
+                    "and will be removed when the JVM is shutdown, file: {}", temp.toString());
+            temp.toFile().deleteOnExit();
+        }
+    }
+
+    private static void writeJsonObjectToFile(Path temp, Path output, Object json) throws IOException {
+        // TODO
+        //  To fix defect Trello #616 I've refactored this code to ensure that the temp file is deleted immeditely
+        //  after use.
+        //  I am not sure why this first step of serialising to a temp file is required it seems unncessary to me?
+        //  However, this functionality is it at the centre of the website and publishing services so I've made a
+        //  concious tactical decision to only fix the immediate problem of ensuring the temp files are deleted rather
+        //  than rewriting/refactoring this whole process and risk introducing a regression.
```

Comment:
A few more typos sorry. Also wonder if it's worth, changing the perspective from "I" to a neutral non-poseesive language ("I am not sure…" -> "It is not clear…" etc.). 
```suggestion
        // TODO
        //  To fix defect Trello #616 I've refactored this code to ensure that the temp file is deleted immediately
        //  after use.
        //  I am not sure why this first step of serialising to a temp file is required it seems unnecessary to me?
        //  However, this functionality is it at the centre of the website and publishing services so I've made a
        //  conscious tactical decision to only fix the immediate problem of ensuring the temp files are deleted rather
        //  than rewriting/refactoring this whole process and risk introducing a regression.
```